ports/downwell/Downwell.sh

```diff
@@ -0,0 +1,107 @@
+
+# check if we have new enough version of PortMaster that contains xdelta3
+if [ ! -f "$controlfolder/xdelta3" ]; then
+  echo "This port requires the latest PortMaster to run, please go to https://portmaster.games/ for more info." > /dev/tty0
+  sleep 5
+  exit 1
+fi
+
+# Patch game
+cd "$GAMEDIR"
+
+# If "gamedata/data.win" exists and matches the checksum of the itch or steam versions
+if [ -f "./gamedata/data.win" ]; then
+    checksum=$(md5sum "./gamedata/data.win" | awk '{print $1}')
+    
+    # Checksum for the GOG version
+    if [ "$checksum" = "fbecdea0ad4ce643e627fc9aca8e9841" ]; then
+        $ESUDO $controlfolder/xdelta3 -d -s gamedata/data.win -f ./patch/downwellgog.xdelta gamedata/game.droid && \
+        rm gamedata/data.win
+    # Checksum for the Steam version
+    elif [ "$checksum" = "3ca6acd5af13997786e27bb5694fb103" ]; then
+        $ESUDO $controlfolder/xdelta3 -d -s gamedata/data.win -f ./patch/downwellsteam.xdelta gamedata/game.droid && \
+        rm gamedata/data.win
+    else
+        echo "Error: MD5 checksum of game.unx does not match any expected version."
+    fi
+else    
+    echo "Error: Missing files in gamedata folder or game has been patched."
+fi
+
+# Pack Audio into apk and move game files to the right place
+if [ -n "$(ls ./gamedata/*.dat 2>/dev/null)" ]; then
+    # Move all audiogroup.dat from ./gamedata to ./assets
+    mkdir -p ./assets
+    mv ./gamedata/*.dat ./assets/ || exit 1
+    mv ./gamedata/*.ogg ./assets/ || exit 1
+    echo "Moved audiogroup.dat files from ./gamedata to ./assets/"	
+
+    # Zip the contents of ./game.apk including the new .ogg and .wav files
+    zip -r -0 ./game.apk ./assets/ || exit 1
+    echo "Zipped contents to ./game.apk"
+    rm -Rf "$GAMEDIR/assets/" || exit 1
+
+    # cleanup if extra files were copied in from steam or gog
+    rm -Rf "$GAMEDIR/gamedata/Downwell.exe" \
+           "$GAMEDIR/gamedata/"*.dll \
+           "$GAMEDIR/gamedata/"*.ico \
+           $GAMEDIR/gamedata/gog* \
+           $GAMEDIR/gamedata/unins000*
+
+    # Move all files from gamedir/gamedata to gamedir and delete gamedata folder
+    mv "$GAMEDIR/gamedata"/* "$GAMEDIR" && rm -rf "$GAMEDIR/gamedata" || exit 1
+    echo "Moving and Cleaning of game files done"
+fi
+
+# Check if CFW_NAME is 'ArkOS' or 'ArkOS wuMMLe' to patch libzip.so.5
+if [ "$CFW_NAME" == "ArkOS" ] || [ "$CFW_NAME" == "ArkOS wuMMLe" ]; then
+    # Replace libzip.so.5 in libs folder with libzip.so.5.arkos
+    if [ -f "$GAMEDIR/libs/libzip.so.5" ] && [ -f "$GAMEDIR/patch/libzip.so.5.arkos" ]; then
+        rm "$GAMEDIR/libs/libzip.so.5" || { echo "Failed to delete libzip.so.5"; exit 1; }
+        cp "$GAMEDIR/patch/libzip.so.5.arkos" "$GAMEDIR/libs/libzip.so.5" || { echo "Failed to replace libzip.so.5"; exit 1; }
+        echo "libzip.so.5 replaced successfully."
+    else
+        echo "Error: libzip.so.5 or libzip.so.5.arkos not found."
+        exit 1
+    fi
+fi
```

Comment:
```suggestion
```
A couple things: For one, this isn't a patch; these are two complete and separate files and this copies one of those in place of the other if ArkOS is detected - no patching is being done to either of those files.

The more important question I have is... is this necessary?  Is there a reason we don't just have the libzip.so.5 that works with ArkOS as the singular provided .so file since that should work across all OSes without "juggling" dependencies?  This would make things simpler and this shell script even shorter.

Reply:
Will try to test this... but the original .so.5 works with  Rocknix/MuOS and NOT ArkOS while the OS replacement works only for ArkOS. I am not sure why this happens though. Can test more later this week when I am free